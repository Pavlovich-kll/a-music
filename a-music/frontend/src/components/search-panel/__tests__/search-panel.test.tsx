import React from 'react';
import SearchPanel from '../index';
import { shallow } from 'enzyme';
import AsyncSelect from 'react-select/async';

const mockHandleChange = jest.fn();
let wrapper: any;

describe('SearchPanelShallow', () => {
  beforeEach(() => {
    wrapper = shallow(<SearchPanel setToPlay={mockHandleChange} />);
  });

  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should trigger setToPlay function', () => {
    const select = wrapper.find(AsyncSelect);
    select.props().onChange({ value: 'some value' });
    expect(mockHandleChange).toBeCalled();
  });
});
