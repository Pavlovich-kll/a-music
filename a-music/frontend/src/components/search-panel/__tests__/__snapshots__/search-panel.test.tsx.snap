// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanelShallow should render 1`] = `
<styled.div>
  <Async
    autoFocus={true}
    cacheOptions={true}
    defaultOptions={false}
    filterOption={null}
    inputValue=""
    isClearable={true}
    isLoading={false}
    loadOptions={[Function]}
    onChange={[Function]}
    onInputChange={[Function]}
    placeholder="Search..."
    styles={
      Object {
        "clearIndicator": [Function],
        "dropdownIndicator": [Function],
        "option": [Function],
        "valueContainer": [Function],
      }
    }
    value=""
  />
</styled.div>
`;
